src/ripple/app/main/LoadManager.cpp

```diff
@@ -32,38 +33,50 @@
 
 namespace ripple {
 
-class LoadManagerImp
-    : public LoadManager
-    , public beast::Thread
+class LoadManagerImp : public LoadManager
 {
 public:
     //--------------------------------------------------------------------------
 
-    beast::Journal m_journal;
-    using LockType = std::mutex;
-    using ScopedLockType = std::lock_guard <LockType>;
-    LockType mLock;
+    beast::Journal mJournal;
+    std::thread mThread;
 
-    bool mArmed;
+    std::mutex mMutex;          // Guards mDeadLock and mArmed
 
     int mDeadLock;              // Detect server deadlocks
+    bool mArmed;
+
     //--------------------------------------------------------------------------
 
     LoadManagerImp (Stoppable& parent, beast::Journal journal)
         : LoadManager (parent)
-        , Thread ("loadmgr")
-        , m_journal (journal)
-        , mArmed (false)
+        , mJournal (journal)
+        , mThread ()
+        , mMutex ()
         , mDeadLock (0)
+        , mArmed (false)
     {
         UptimeTimer::getInstance ().beginManualUpdates ();
     }
 
-    ~LoadManagerImp ()
+    LoadManagerImp () = delete;
+    LoadManagerImp (LoadManagerImp const&) = delete;
+    LoadManagerImp& operator=(LoadManager const&) = delete;
+
+    ~LoadManagerImp () override
     {
-        UptimeTimer::getInstance ().endManualUpdates ();
+        try
+        {
+            UptimeTimer::getInstance ().endManualUpdates ();
 
-        stopThread ();
+            if (mThread.joinable())
+                onStop ();
+        }
+        catch (...)
```

Comment:
If we caught both `std::exception` and `...` we could write `what()` to the log. Could be useful for debugging.